subtrack.MAUI/Services/SubscriptionService.cs

```diff
@@ -14,5 +15,21 @@ public async Task<IEnumerable<Subscription>> GetSubscriptions()
         {
             return await _context.Subscriptions.ToListAsync();
         }
+
+        public async Task Update(Subscription subscriptionToUpdate)
+        {
+            var sub = await _context.Subscriptions.FindAsync(subscriptionToUpdate.Id);
+
+            if (sub != null)
```

Comment:
either do `if(sub == null) throw new NotFoundException()` or simply ignore the fact that there is a 0.01 chance that sub is null in our case (since we don't have concurrent users)